
From: Bjorn Helgaas <bjorn.helgaas@hp.com>

Make assign_irq_vector() non-__init always (it's called from
io_apic_set_pci_routing(), which is used in the pci_enable_device() path).

Signed-off-by: Bjorn Helgaas <bjorn.helgaas@hp.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/arch/i386/kernel/io_apic.c |    4 ----
 1 files changed, 4 deletions(-)

diff -puN arch/i386/kernel/io_apic.c~make-assign_irq_vector-non-__init arch/i386/kernel/io_apic.c
--- 25/arch/i386/kernel/io_apic.c~make-assign_irq_vector-non-__init	2004-08-28 14:34:16.936760168 -0700
+++ 25-akpm/arch/i386/kernel/io_apic.c	2004-08-28 14:34:16.942759256 -0700
@@ -1120,11 +1120,7 @@ static inline int IO_APIC_irq_trigger(in
 /* irq_vectors is indexed by the sum of all RTEs in all I/O APICs. */
 u8 irq_vector[NR_IRQ_VECTORS] = { FIRST_DEVICE_VECTOR , 0 };
 
-#ifdef CONFIG_PCI_MSI
 int assign_irq_vector(int irq)
-#else
-int __init assign_irq_vector(int irq)
-#endif
 {
 	static int current_vector = FIRST_DEVICE_VECTOR, offset = 0;
 
_
